decentraland/components/presentational/BuyLand.test.js

```diff
@@ -4,18 +4,115 @@ import BuyLand from "./BuyLand";
 import { parcelForSale } from "./testHelpers";
 
 describe("BuyLand", () => {
-  it("renders the component", async () => {
+  describe("renders the component", () => {
     const onBuy = () => {};
-    const waitingForAccountSetup = false;
 
-    expect(
-      shallow(
-        <BuyLand
-          landForSale={parcelForSale}
-          onBuy={onBuy}
-          waitingForAccountSetup={waitingForAccountSetup}
-        />
-      )
-    ).toMatchSnapshot();
+    it("initial state - before start account setup", async () => {
+      const accountInfo = {
+        setupInProgress: false,
+        fundedWithEthers: false,
+        fundedWithMana: false,
+        approvedMarketplace: false,
+      };
+      expect(
+        shallow(
+          <BuyLand
+            landForSale={parcelForSale}
+            onBuy={onBuy}
+            accountInfo={accountInfo}
+          />
+        )
+      ).toMatchSnapshot();
+    });
+
+    it("waiting for setup - waiting for ETH funding", async () => {
+      const accountInfo = {
+        setupInProgress: true,
+        fundedWithEthers: false,
+        fundedWithMana: false,
+        approvedMarketplace: false,
+      };
+      expect(
+        shallow(
+          <BuyLand
+            landForSale={parcelForSale}
+            onBuy={onBuy}
+            accountInfo={accountInfo}
+          />
+        )
+      ).toMatchSnapshot();
+    });
+
+    it("waiting for setup - waiting for MANA funding and marketplace approval", async () => {
+      const accountInfo = {
+        setupInProgress: true,
+        fundedWithEthers: true,
+        fundedWithMana: false,
+        approvedMarketplace: false,
+      };
+      expect(
+        shallow(
+          <BuyLand
+            landForSale={parcelForSale}
+            onBuy={onBuy}
+            accountInfo={accountInfo}
+          />
+        )
+      ).toMatchSnapshot();
+    });
+
+    it("waiting for setup - waiting for MANA funding", async () => {
+      const accountInfo = {
+        setupInProgress: true,
+        fundedWithEthers: true,
+        fundedWithMana: false,
+        approvedMarketplace: true,
```

Comment:
I guess the fact that this can happen in either order with the MANA funding is the only potential hiccup for the enumerable list of strings idea

Reply:
Following the new convention, "this" here refers to the lowest line of code shown, the `approvedMarketplace` one

Reply:
The parallelism adds more complexity, because of that I've created the message "Funding account with MANA and approving Marketplace..."

Reply:
That makes sense. We can resolve this then.

Reply:
Resolving this.